contracts/MechMarketplace.sol

```diff
@@ -381,132 +376,268 @@ contract MechMarketplace is IErrorsMarketplace {
+        bytes memory paymentData
+    ) external payable returns (uint256[] memory requestIds) {
+        // Reentrancy guard
+        if (_locked) {
+            revert ReentrancyGuard();
+        }
+        _locked = true;
+
+        requestIds = _requestBatch(requestDatas, priorityMechServiceId, requesterServiceId, responseTimeout, paymentData);
+
+        _locked = false;
+    }
+
+    /// @dev Delivers requests.
     /// @notice This function can only be called by the mech delivering the request.
-    /// @param requestId Request id.
-    /// @param deliveryData Self-descriptive opaque delivery data-blob.
-    function deliverMarketplace(uint256 requestId, bytes memory deliveryData) external {
+    /// @param requestIds Set of request ids.
+    /// @param deliveryRates Corresponding set of actual charged delivery rates for each request.
+    /// @param deliveryDatas Set of corresponding self-descriptive opaque delivery data-blobs.
+    function deliverMarketplace(
+        uint256[] memory requestIds,
+        uint256[] memory deliveryRates,
+        bytes[] memory deliveryDatas
+    ) external returns (bool[] memory deliveredRequests) {
         // Reentrancy guard
-        if (_locked > 1) {
+        if (_locked) {
             revert ReentrancyGuard();
         }
-        _locked = 2;
+        _locked = true;
+
+        // Check array lengths
+        if (requestIds.length == 0 || requestIds.length != deliveryRates.length ||
+            requestIds.length != deliveryDatas.length) {
+            revert WrongArrayLength3(requestIds.length, deliveryRates.length, deliveryDatas.length);
+        }
 
         // Check delivery mech and get its service multisig
         address mechServiceMultisig = checkMech(msg.sender);
 
-        // Get mech delivery info struct
-        MechDelivery storage mechDelivery = mapRequestIdDeliveries[requestId];
-        address priorityMech = mechDelivery.priorityMech;
+        uint256 numDeliveries;
+        uint256 numRequests = requestIds.length;
+        // Allocate delivered requests array
+        deliveredRequests = new bool[](numRequests);
+        // Allocate requester related arrays
+        address[] memory requesters = new address[](numRequests);
+        uint256[] memory requesterDeliveryRates = new uint256[](numRequests);
+
+        // Get delivery mech payment type
+        bytes32 paymentType = IMech(msg.sender).paymentType();
+
+        // Traverse all requests being delivered
+        for (uint256 i = 0; i < numRequests; ++i) {
+            // Get request info struct
+            RequestInfo storage requestInfo = mapRequestIdInfos[requestIds[i]];
+            address priorityMech = requestInfo.priorityMech;
+
+            // TODO continue instead of revert? This would let us just ignore incorrect requests without reverting the bulk
+            // TODO Additional thought: leave revert as it would stop from mech trying to request arbitrary requests and not actually recorded ones
```

Comment:
What scenario is this?

Reply:
Mech is trying to deliver request that was never registered on the MM